code/game/gamemodes/meteor/meteors.dm

```diff
@@ -170,7 +171,8 @@
 	icon_state = "medium_radioactive"
 
 /obj/effect/meteor/radioactive/Bump(atom/a)
-
+	if(loc == null)
```

Comment:
Couldn't you just have each child of meteor inherit the bump from the parent object, save copypasting the check, or does that not work with effects?